BoostConfig.cmake

```diff
@@ -153,6 +159,63 @@ macro(boost_find_component comp req)
 
 endmacro()
 
+macro(boost_find_all_components req)
+
+  if(Boost_DEBUG)
+    if(${req})
+      message(STATUS "BoostConfig: All components are required.")
+    else()
+      message(STATUS "BoostConfig: All components are optionally requested.")
+    endif()
+  endif()
+
+  # Search for all available component-configuration directories...
+  file(GLOB __boost_all_components
+       LIST_DIRECTORIES true RELATIVE "${CMAKE_CURRENT_LIST_DIR}/.."
+       "${CMAKE_CURRENT_LIST_DIR}/../boost_*-${Boost_VERSION}")
+  # ...and extract component names from it.
+  string(REGEX REPLACE "boost_([_a-z0-9]+)-${Boost_VERSION}" "\\1"
+         __boost_all_components "${__boost_all_components}")
+  list(REMOVE_ITEM __boost_all_components "headers")
+  list(LENGTH __boost_all_components __boost_all_components_count)
+
+  if(${__boost_all_components_count} EQUAL 0 AND ${req})
+    message(SEND_ERROR "BoostConfig: No components available (although all are required)!")
+  endif()
+
+  if(Boost_DEBUG AND Boost_VERBOSE)
+    if(${__boost_all_components_count} EQUAL 0)
+      message(STATUS "BoostConfig: Following components will be searched: N/A")
+    else()
+      message(STATUS "BoostConfig: Following components will be searched:")
+      foreach(__boost_comp IN LISTS __boost_all_components)
+        message(STATUS "  ${__boost_comp}")
+      endforeach()
+    endif()
+  endif()
+
+  # Try to find each component.
+  foreach(__boost_comp IN LISTS __boost_all_components)
+
+    boost_find_component(${__boost_comp} ${req})
+
+    # Append to list of all targets (if found).
+    string(TOUPPER ${__boost_comp} _BOOST_COMP)
+    if(Boost_${_BOOST_COMP}_FOUND)
+      list(APPEND Boost_ALL_TARGETS Boost::${__boost_comp})
+    endif()
+
+  endforeach()
+
+
+  #  list(REMOVE_DUPLICATES Boost_ALL_TARGETS)
```

Comment:
I just realized, that these lines could be removed.  
I will do so before this pull-request will be accepted/merged.